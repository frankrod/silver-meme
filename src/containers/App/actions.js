import * as ActionTypes from './constants';

export function getImages() {
  return {
    type: ActionTypes.GET_IMAGES
  };
}

export function getImagesSuccess(images) {
  return {
    type: ActionTypes.GET_IMAGES_SUCCESS,
    payload: images
  };
}

export function getImagesFail() {
  return {
    type: ActionTypes.GET_IMAGES_FAIL
  };
}