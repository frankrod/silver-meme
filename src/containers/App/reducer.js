import * as ActionTypes from './constants';

const initialState = {
  images: []
};

const appReducer = (state = initialState, action) => {
  const { type, payload } = action;
  switch (type) {
    case ActionTypes.GET_IMAGES: {
      return { ...state, images: payload };
    }
    default:
      return state;
  }
};

export default appReducer;
