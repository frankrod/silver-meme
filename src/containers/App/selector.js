import { createSelector } from 'reselect';
import { isNil } from 'ramda';

const getAppState = (app) => app;

export const getTotalPages = createSelector(
  getAppState, 
  ({ images }) => {
    if (isNil(images)) return 0;

    return Math.ceil(images.photos.total / images.photos.perpage);
  },
);
