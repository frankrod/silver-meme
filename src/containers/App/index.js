import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as appActions from './actions';
import logo from '../../images/logo.svg';
import './App.css';

class App extends Component {
  componentDidMount() {
    this.props.actions.getImages();
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
      </div>
    );
  }
}

export const mapStateToProps = ({ app } ) => {
  return {
    images: app.images,
  };
};

export function mapDispatchToProps(dispatch: any): Object {
  const actions = bindActionCreators(
    { ...appActions },
    dispatch
  );
  return { actions };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);

