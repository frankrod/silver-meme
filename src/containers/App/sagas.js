import { put, takeLatest } from 'redux-saga/effects';

import * as ActionTypes from './constants';
import * as appActions from './actions';

export function* fetchImages(action: Action<string>): Saga<void> {
  try {    
    yield put(appActions.getImagesSuccess({image: ''}));
  } catch (err) {
    yield put(appActions.getImagesFail(err));
  }
}

export function* watchGetImages(): Saga<void> {
  yield takeLatest(ActionTypes.GET_IMAGES, fetchImages);
}

export const appSagas = [
  watchGetImages
];
