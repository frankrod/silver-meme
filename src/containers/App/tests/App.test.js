import React from 'react';
import { render } from 'react-testing-library'
import App from '../index';

it('renders without crashing', () => {
  const { container } = render(<App />)

  expect(container).toMatchSnapshot()
});
