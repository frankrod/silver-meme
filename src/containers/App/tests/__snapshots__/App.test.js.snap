// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        src="logo.svg"
        class="App-logo"
        alt="logo"
      />
      <h1
        class="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <p
      class="App-intro"
    >
      To get started, edit 
      <code>
        src/App.js
      </code>
      and save to reload.
    </p>
  </div>
</div>
`;
